data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/BeaconRestApi.java

```diff
@@ -388,6 +390,9 @@ private void addValidatorHandlers(final DataProvider dataProvider, final Spec sp
     app.post(
         PostContributionAndProofs.ROUTE, new PostContributionAndProofs(dataProvider, jsonProvider));
     addMigratedEndpoint(new PostPrepareBeaconProposer(dataProvider));
+    if (spec.isMilestoneSupported(SpecMilestone.BELLATRIX)) {
+      addMigratedEndpoint(new PostRegisterValidator(dataProvider, spec, schemaCache));
+    }
```

Comment:
I suspect we should always add the endpoint but then just return bad request if bellatrix isn't enabled. It's weird to get a 404 based on the network config.